deleting thread now redirects to discussions

Israellund opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #8744 

## Description of Changes
- User is now redirected to `/discussions` after deleting a thread from within the thread

## "How We Fixed It"
<!-- Brief description of solution, if bug, to be added once issue is resolved. -->
-created a new function `getThreadUrlPath` to replace the use of `getProposalUrlPath` that redirects to `/discussions` instead of `/proposal/${id}` if `type === ProposalType.Thread` is false
NOTE: I changed `getProposalUrlPath` to `getThreadUrlPath` in `DiscussionsPage.tsx` as well as `ViewThreadPage.tsx` to stay consistent with coding intent, despite it not being needed in `DiscussionsPage.tsx` 
## Test Plan
- go to a thread you create or create a thread
- while on `/discussions` page delete the thread
- confirm that thread was deleted and that you are still on `/discussions`
- go to another thread you created or create a new thread
- click into that thread and delete the thread from there
- confirm that you are now redirected to `/discussions`

jnaviask commented on packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:
```diff
@@ -127,6 +127,29 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
+    if (
+      !isLoadingTopics &&
+      decodedString &&
+      splitURLPath[2] === 'discussions' &&
```
Did we test this piece against custom domains?

Israellund replied:
@jnaviask This should be fine for custom domains as the url is being split on "/", so there should be no difference.

masvelio replied:
in https://gov.1inch.io/discussions 
![image](https://github.com/user-attachments/assets/4c3e1d3a-d89a-436e-9fd9-45559bebb3a5)


in https://commonwealth.im/1inch/discussions
![image](https://github.com/user-attachments/assets/387eb72d-6ff5-4b37-9286-d16446453a6a)

Israellund replied:
Whoops. Okay, I was wrong, I will fix that.

masvelio commented on packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:
```diff
@@ -127,6 +127,29 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
+      splitURLPath[2] === 'discussions' &&
+      splitURLPath.length === 4
+    ) {
+      const validTopics = topics?.some(
```
why you check topics here? The bug is about deleted thread, not topic.

Israellund replied:
I don't know why this code was included in this PR, this is from a different PR I had made to redirect users if they input the incorrect url that was merged already. But I will now change it here to account for custom domains, as mentioned above.

masvelio reviewed (changes requested):
infinite loader after visiting deleted thread 
https://github.com/user-attachments/assets/711a3159-f18c-4c2b-98d7-4760ebe535bc

This fix seems to be unrelated to the bug. We should check if thread data is fetched in ViewThreadPage. If not, redirect the user

Israellund commented:
@masvelio I changed the logic to handle custom urls and pushed that change. I wasn't sure how to replicate what you were showing in your loom, as that is what was shown before when a thread was deleted from inside the thread. I tried to create a thread, copy the url, delete the thread, then paste in the now deleted thread url, but it just shows "Thread not found". It also shows this if I delete the topic name after the "-" 
Please let me know if I'm missing something in regards to your loom though.

burtonator commented:
@Israellund I went ahead and approved but there is this issue which is further complicated by this fix:

https://github.com/hicommonwealth/commonwealth/issues/8853

Basically we have too many ways to redirect users to URLs and half of them are inconsistent.

Israellund commented:
@masvelio I used  // eslint-disable-next-line react-hooks/exhaustive-deps for the useEffect dependency array because if navigate is included in it it causes an infinite rerender. You commented on another PR that we shouldn't be using this, so let me know if you'd like for me to change it at all. This was the best way I could find to fix this.

Israellund commented:
This PR handles the bad url redirect better as it accounts for custom domains, therefore I am closing [this PR](https://github.com/hicommonwealth/commonwealth/pull/8729)

Israellund commented:
Closing this PR. The bug found when deleting a thread was fixed in #8248 and the custom domains fix mentioned in the comments above is fixed in PR #8928
